library/kani/src/invariant.rs

```diff
@@ -0,0 +1,13 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+//! This module introduces the Invariant trait.
+
+/// This trait should be used to specify and check type safety invariants for a
+/// type.
+pub trait Invariant
+where
+    Self: Sized,
+{
+    fn is_valid(&self) -> bool;
```

Comment:
I would like to suggest `is_safe(&self)`... what do you think?

Reply:
It's a great suggestion since a lot of related contents refers to this as "type safety". Changed to `is_safe`.